### PR TITLE
Use FactoryBot to generate params in the create_spec

### PR DESCRIPTION
**Pull request by sideshowbandana**
If FactoryBot exists, use it to generate the params for the basic spec

### CODE REVIEW

File: lib/generators/graphiti/templates/create_request_spec.rb.erb [hunk 1]
```diff
@@ -7,14 +7,14 @@ RSpec.describe "<%= type %>#create", type: :request do
 
   describe 'basic create' do
     let(:params) do
-      <%- if defined?(FactoryBot) %>
-      FactoryBot.build(:<%= type.to_s.singularize %>).attributes.
-      except("created_at", "updated_at")
-      <%- else %>
+      <%- if defined?(FactoryBot) -%>
+      build(:<%= type.to_s.singularize %>).attributes.
```
Comment (richmolj) [inline lib/generators/graphiti/templates/create_request_spec.rb.erb:11]: I think we can use `attributes_for` to get only the attributes specified in the factory

**Review by richmolj — COMMENTED** [review]
Quick comment. FWIW I've been leaning against doing this to force a little more thought by the developer, but I think I'm turning around. @wadetandy any opinion?

**Comment by wadetandy** [comment]
I'm conflicted on this change.  On the one hand, for a large swath of features (especially starting out), one's API resources are going to largely map to underlying models (and therefore factories) 1:1.  This feature makes it easier to bootstrap new graphiti resources (and their specs) and get up and running quickly.  On the other hand, one of the more valuable concepts of graphiti and the Resource is enabling the decoupling of the model from the API. Using factory_bot for APIs by default might bury this concept more than we'd like.  

I can see a couple paths here: one which we should probably do as a command line option for the generator is the ability to either enable or disable this, depending on which default we go with.  This should probably happen either way.  The other is to introspect whether the factory for the model exists before adding this behavior to a given spec.  This will serve as a reasonable facsimile for whether a Model and API have a 1:1 mapping and should have this as default bootstrap behavior.

**Comment by richmolj** [comment]
@wadetandy I agree with the premise of separating DB and API, but there's already somewhat loose coupling between a **model** and API. FactoryBot can be used without ActiveRecord, and should hold only attributes to pass validation (more or less) so it makes sense they would hold attributes needed for `create`.

> The other is to introspect whether the factory for the model exists before adding this behavior to a given spec

The generators actually already assume ActiveRecord and factories - they are very much the "Rails defaults happy path". It would be interesting to be more generic, but I'm not totally sure that's even possible. I agree with your general premise but I think this PR honors the current status quo.

**Comment by sideshowbandana** [comment]
I agree with what you guys are saying here. The reason that I threw this together was because I wanted to bootstrap a bunch of classes and then number of failures that I was getting in the tests due to the lack of definition in the Resources was too much to deal with while I was changing around the schema and prototyping.

I think that the right approach would be to allow the user to explicitly ask for the `default_attributes` to be used. And make a note as part of it that the scaffolding should be updated for a production system.

**Comment by wadetandy** [comment]
Agree with your reasoning that these generators are already assuming an active record context so I’m fine with this change with the understanding we would like to eventually make it more generic & configurable.

**Comment by richmolj** [comment]
@sideshowbandana how about

```bash
$ bin/rails g graphiti:resource Post attrs:Post
```

Where the `:Post` is ideally optional. The thought being, this would accommodate your use case without a DSL change, and without the downside of a model adding columns later that shouldn't be exposed.

> fine with this change with the understanding we would like to eventually make it more generic & configurable.

Spiritually yes we share the same goal. I think it is pretty hard to do in practice, but agree this would be better/more correct.

**Comment by richmolj** [comment]
Whoops, I think this PR is getting a little comingled with https://github.com/graphiti-api/graphiti/pull/76

For this one, I think we can use `attribute_for` and merge

**Comment by sideshowbandana** [comment]
> For this one, I think we can use `attribute_for` and merge

`attributes_for` falls short when there's an association dependency. FWICT attributes_for can only create the attributes but not the associations.

**Comment by sideshowbandana** [comment]
PR moved to new repo

**Pull request closed without merging**